.github/workflows/go.yml

```diff
@@ -0,0 +1,28 @@
+name: lint-and-test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  lint-test-and-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+      - name: lint
+        uses: golangci/golangci-lint-action@v3.3.1
+        with:
+          github-token: ${{ github.token }}
+
+      - name: test
+        run: go test -v -race ./...
+
+      - name: build
```

Comment:
This is probably unnecessary since there's no `package main` binary to be built and the package will be compiled anyway as long as there is at least 1 test.

Reply:
removed it, thanks